src/main/java/com/google/cloud/tools/project/AppYaml.java

```diff
@@ -16,54 +16,113 @@
 
 package com.google.cloud.tools.project;
 
-import java.io.IOException;
 import java.io.InputStream;
-import java.nio.file.Files;
-import java.nio.file.Path;
 import java.util.Collections;
 import java.util.Map;
 import javax.annotation.Nullable;
 import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.SafeConstructor;
 
-/** Tools for reading app.yaml */
+/** Tools for reading {@code app.yaml}. */
 public class AppYaml {
 
-  private final Map<String, ?> yamlMap;
-
+  private static final String ENVIRONMENT_TYPE_KEY = "env";
   private static final String RUNTIME_KEY = "runtime";
+  private static final String API_VERSION_KEY = "api_version";
+  private static final String APPLICATION_KEY = "application";
+  private static final String VERSION_KEY = "version";
+  private static final String SERVICE_KEY = "service";
+  private static final String MODULE_KEY = "module";
+  private static final String ENVIRONMENT_VARIABLES_KEY = "env_variables";
+
+  private final Map<String, ?> yamlMap;
 
   /**
    * Parse an app.yaml file to an AppYaml object.
    *
-   * @param appYaml the app.yaml file
-   * @throws IOException if reading app.yaml fails due to I/O errors
+   * @param input the input, typically the contents of an {@code app.yaml} file
    * @throws org.yaml.snakeyaml.scanner.ScannerException if reading app.yaml fails while scanning
    *     due to malformed YAML (undocumented {@link RuntimeException} from {@link Yaml#load})
    * @throws org.yaml.snakeyaml.parser.ParserException if reading app.yaml fails while parsing due
```

Comment:
ditto this one. Perhaps not in this PR, but we do need to replace these with a checked exception from our own API. SnakeYAML should not be part of our interface contract. A custom subclass of IOException might be appropriate.